### PR TITLE
[Windows] Fix SysApp CPU load API.

### PR DESCRIPTION
**Pull request by darktears**
Implement getting the CPU load from the performance counters API.
This API needs to be initialized so I've added an init() function
which is used only on Windows (other platforms are not doing anything).

### CODE REVIEW

File: sysapps/device_capabilities/cpu_info_provider.h [hunk 1]
```diff
@@ -30,6 +30,8 @@ class CPUInfoProvider {
   // Android (via /proc/loadavg).
   double GetCPULoad() const;
 
+  void init();
```
Comment (rakuco) [inline sysapps/device_capabilities/cpu_info_provider.h]: How about making it `virtual` and providing an empty implementation here?

Comment (darktears) [inline sysapps/device_capabilities/cpu_info_provider.h]: then each backend needs to subclass this which is not the case today.

Comment (rakuco) [inline sysapps/device_capabilities/cpu_info_provider.h]: Ah, I didn't see those were not subclasses.

Comment (darktears) [inline sysapps/device_capabilities/cpu_info_provider.h]: :) I would have gone for the virtual otherwise :)

Comment (pozdnyakov) [inline sysapps/device_capabilities/cpu_info_provider.h]: I'm afraid that wouldn't work as invoked from the ctor :)

File: sysapps/device_capabilities/cpu_info_provider_win.cc [hunk 2]
```diff
@@ -4,14 +4,27 @@
...
 #include "xwalk/sysapps/device_capabilities/cpu_info_provider.h"
 
+#include <pdh.h>
+
 #include "base/logging.h"
 
 namespace xwalk {
 namespace sysapps {
 
+static PDH_HQUERY cpuQuery;
```
Comment (pozdnyakov) [inline sysapps/device_capabilities/cpu_info_provider_win.cc]: nameless namespace?

File: sysapps/device_capabilities/cpu_info_provider_win.cc [hunk 3]
```diff
@@ -4,14 +4,27 @@
...
+static PDH_HCOUNTER cpuTotal;
```
Comment (pozdnyakov) [inline sysapps/device_capabilities/cpu_info_provider_win.cc]: ditto

File: sysapps/device_capabilities/cpu_info_provider_win.cc [hunk 4]
```diff
@@ -4,14 +4,27 @@
...
+
+void CPUInfoProvider::init() {
+  PdhOpenQuery(NULL, NULL, &cpuQuery);
```
Comment (pozdnyakov) [inline sysapps/device_capabilities/cpu_info_provider_win.cc]: nit: nullptr

File: sysapps/device_capabilities/cpu_info_provider_android.cc [hunk 5]
```diff
@@ -20,6 +20,8 @@ const char kProcLoadavg[] = "/proc/loadavg";
 namespace xwalk {
 namespace sysapps {
 
+void CPUInfoProvider::init() {}
```
Comment (pozdnyakov) [inline sysapps/device_capabilities/cpu_info_provider_android.cc]: do we anticipate anything here? maybe just put the stuff to the constructor in the windows-specific class (not introduce init() method)?

File: sysapps/device_capabilities/cpu_info_provider_win.cc (hunk 2 above)
Comment (halton) [inline sysapps/device_capabilities/cpu_info_provider_win.cc]: Do we really needed anonymous? `cpuQuery` is not declared in header. BTW, `cpuQeury` -> `cpu_query`

Comment (darktears) [inline sysapps/device_capabilities/cpu_info_provider_win.cc]: Anonymous is apparently the pattern in Chromium even in .cc files. Will rename it though.

File: sysapps/device_capabilities/cpu_info_provider.h (hunk 1 above)
Comment (darktears) [inline sysapps/device_capabilities/cpu_info_provider.h]: Yup.

File: sysapps/device_capabilities/cpu_info_provider_win.cc (hunk 4 above)
Comment (darktears) [inline sysapps/device_capabilities/cpu_info_provider_win.cc]: only some can be replaced, the second NULL here is a DWORDPTR and thus doesn't compile with nullptr. Not sure I want to add a cast here.

File: sysapps/device_capabilities/cpu_info_provider_android.cc (hunk 5 above)
Comment (pozdnyakov) [inline sysapps/device_capabilities/cpu_info_provider_android.cc]: so, would it be simpler just to add functional code to the Win-specific constructor?

Comment (darktears) [inline sysapps/device_capabilities/cpu_info_provider_android.cc]: Well there is no Win-specific constructor, the _win file is not a subclass if CPUInfoProvider. So what I can do is ifdef out the constructer in cpu_info_provider.cc only for Windows and implement the CPUInfoProvider constructor in the _win class. The other solution is to put the few init() lines inside cpu_info_provider.cc directly protected by a ifdef only on Windows, but then again what's the point of a _win file in the first place.

File: sysapps/device_capabilities/cpu_info_provider_win.cc [hunk 6]
```diff
@@ -4,14 +4,30 @@
...
 
+#include <pdh.h>
+
 #include "base/logging.h"
 
 namespace xwalk {
 namespace sysapps {
 
+namespace {
+  static PDH_HQUERY cpu_query;
```
Comment (pozdnyakov) [inline sysapps/device_capabilities/cpu_info_provider_win.cc]: 'static' can be omitted now.

File: sysapps/device_capabilities/cpu_info_provider_android.cc (hunk 5 above)
Comment (pozdnyakov) [inline sysapps/device_capabilities/cpu_info_provider_android.cc]: ah, I see. I'm OK with 'init()' than

**Comment by pozdnyakov** [comment]
lgtm

**Pull request merged**